### PR TITLE
Fix the remaining date fields that have issues.

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

I fixed most of these fields in #149

## Purpose/Implementation Notes

@davidsmejia pointed out that embargo_date was still a TextField. I think after this everything is well though!

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

Populating the database with dev data works smoothly!

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: api/resources_portal/management/commands/populate_test_database.py [hunk 1]
```diff
@@ -27,6 +27,8 @@ def add_class_to_database(class_json, Class):
                     element["updated_at"] = dateparse.parse_datetime(element["updated_at"])
                 if "created_at" in element:
                     element["created_at"] = dateparse.parse_datetime(element["created_at"])
+                if "date_joined" in element:
+                    element["date_joined"] = dateparse.parse_datetime(element["date_joined"])
```
Comment (davidsmejia) [inline api/resources_portal/management/commands/populate_test_database.py:31]:
```suggestion
                    element["date_joined"] = dateparse.parse_datetime(element["date_joined"])
                if "embargo_date" in element:
                    element["embargo_date"] = dateparse.parse_datetime(element["embargo_date"])
```

File: api/resources_portal/management/commands/populate_test_database.py [hunk 2]
```diff
@@ -27,6 +27,8 @@ def add_class_to_database(class_json, Class):
...
```
Comment (davidsmejia) [inline api/resources_portal/management/commands/populate_test_database.py:32]:
```suggestion
                if "embargo_date" in element:
                    element["embargo_date"] = dateparse.parse_datetime(element["embargo_date"])
```

**Comment by kurtwheeler** [comment]
> Looks good maybe we want to also parse embargo_date?

It is parsed! However because its a DateField instead of a DateTimeField the default parser doesn't trigger a warning about a timezone naive datetime.

File: api/resources_portal/management/commands/populate_test_database.py (hunk 2 above)
Comment (kurtwheeler) [reply api/resources_portal/management/commands/populate_test_database.py:32]: This isn't necessary, see: https://github.com/AlexsLemonade/resources-portal/pull/151#issuecomment-622525988

**Pull request merged**